From 4b0434ae785dd9604d3eb53f0355d49ddfbeb54f Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Mon, 24 Sep 2018 21:13:12 -0400
Subject: [PATCH 23/24] Set sect_sz differently

Device sector size is zero or sector sizes do not match

Change-Id: Id563a61f6327861cc8bba7479d5e118a0efc5805
---
 libdiskconfig/diskconfig.c | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/libdiskconfig/diskconfig.c b/libdiskconfig/diskconfig.c
index e8a98981eb..70ce85c1d6 100644
--- a/libdiskconfig/diskconfig.c
+++ b/libdiskconfig/diskconfig.c
@@ -302,17 +302,11 @@ validate(struct disk_info *dinfo)
             goto fail;
         }
 
-        if (!sect_sz) {
-            ALOGE("Device sector size is zero!");
+        if (!sect_sz || sect_sz != dinfo->sect_size) {
+            ALOGE("Device sector size is zero or sector sizes do not match!");
             goto fail;
         }
 
-        if ((sect_sz != 512) && (sect_sz != 4096)) {
-            ALOGE("Device sector size %d is not supported", sect_sz);
-            goto fail;
-        }
-        dinfo->sect_size = sect_sz;
-
         /* allow the user override the "disk size" if they provided num_lba */
         if (!dinfo->num_lba) {
             if (ioctl(fd, BLKGETSIZE64, &disk_size) < 0) {
-- 
2.17.1

